File: VB/EventToCommandExample/Converters/FocusedColumnChangedEventArgsToFieldNameConverter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports DevExpress.Xpf.Grid
Imports DevExpress.Xpf.Mvvm.UI

Namespace EventToCommandExample.Converters
	Public Class FocusedColumnChangedEventArgsToFieldNameConverter
		Inherits EventArgsConverterBase(Of EventArgs)
		Protected Overrides Function Convert(ByVal e As EventArgs) As Object
			Dim args As CurrentColumnChangedEventArgs = CType(e, CurrentColumnChangedEventArgs)
			If args Is Nothing OrElse args.NewColumn Is Nothing Then
				Return String.Empty
			End If
			Return args.NewColumn.FieldName
		End Function
	End Class
End Namespace

File: CS/EventToCommandExample/Converters/FocusedColumnChangedEventArgsToFieldNameConverter.cs
using System;
using System.Linq;
using DevExpress.Xpf.Grid;
using DevExpress.Xpf.Mvvm.UI;

namespace EventToCommandExample.Converters {
    public class FocusedColumnChangedEventArgsToFieldNameConverter : EventArgsConverterBase<EventArgs> {
        protected override object Convert(EventArgs e) {
            CurrentColumnChangedEventArgs args = (CurrentColumnChangedEventArgs)e;
            if(args == null || args.NewColumn == null) {
                return String.Empty;
            }
            return args.NewColumn.FieldName;
        }
    }
}
